src/SpaceStation.cpp

```diff
@@ -458,132 +484,154 @@ bool SpaceStation::DoShipDamage(Ship *s, Uint32 flags, double relVel)
 	return true;
 }
 
-// XXX SGModel door animation. We have one station (hoop_spacestation) with a
-// door, so this is pretty much based on how it does things. This all needs
-// rewriting to handle triggering animations at waypoints.
-//
-// Docking:
-//   Stage 1 (clearance granted): open
-//           (clearance expired): close
-//   Docked:                      close
-//
-// Undocking:
-//   Stage -1 (LaunchShip): open
-//   Post-launch:           close
-//
+void SpaceStation::SwitchToStage(Uint32 bay, DockStage stage)
+{
+	shipDocking_t &dt = m_shipDocking[bay];
+
+	switch (stage) {
+
+	case DockStage::TOUCHDOWN:
+		{
+			dt.ship->ClearThrusterState();
```

Comment:
Off the top of my head, I believe this indentation doesn't match other switch case statements in the codebase.

Reply:
Done!